### ISSUE TITLE
Wrong error message when user provides correct credentials after too many incorrect attempts

### ISSUE THREAD

**Issue by sasaar** [aws-amplify/amplify-ui-swift-authenticator#92]
Description:
This is more feature request than bug, but after user has entered wrong credentials on login several times and then provides correct credentials, localised message for .notAuthorized is returned:

```
if case .notAuthorized(_, _, _) = error {
    return "authenticator.authError.incorrectCredentials".localized()
}
```

AuthenticatorBaseState still seems to know the exact reason:
<img width="439" alt="Screenshot 2024-08-23 at 14 36 36" src="https://github.com/user-attachments/assets/7fdb3ff4-72f5-4c17-9e49-82211803ffc2">

Would it be possible to get own localized message for login error when password attempts are exceeded?

**Comment by ruisebas** [comment]
Hi @sasaar , thanks for opening this issue. 

Unfortunately Cognito returns a `NotAuthorizedException` error both when the credentials are incorrect and when the attempts have been exceeded. Because of that, we don't have a reliable way of differentiating them either, since the `"Password attempts exceeded"` string you see is also directly coming from the SDK and might change without notice.

I'll tag this as a feature request and we'll discuss if there's any alternative we can offer.
Thanks!

**Comment by sasaar** [comment]
Thanks for the prompt reply.

**Comment by ruisebas** [comment]
We've released version [**1.1.7**](https://github.com/aws-amplify/amplify-ui-swift-authenticator/releases/tag/1.1.7) that now lets you use the `errorMap(_:)` view modifier to provide a new message only for the error you wish to overwrite; if you return `nil` then the Authenticator will handle it as usual.

For example, this would achieve what you wish:
```swift
Authenticator { _ in 
    // ....
}
.errorMap { error in
    // Return custom message when password attempts are exceeded
    if case .notAuthorized(let description, _, _) = error, description == "Password attempts exceeded" {
        return .error(message: "[Your custom message]")
    }

    // Let the Authenticator handle all remaining errors
    return nil
}
```

However, as I said earlier, keep in mind that the description string is coming directly from the Cognito service and that it might change.

**Comment by sasaar** [comment]
Many thanks. We will try this (version 1.1.7) asap.

**Comment by ruisebas** [comment]
Hi @sasaar, we've decided to handle this particular scenario within the Authenticator. It's been released in version [1.1.8](https://github.com/aws-amplify/amplify-ui-swift-authenticator/releases/tag/1.1.8).

You can still use the `errorMap(_:)` function to provide custom error messages, but now the Authenticator should already show a different error message when the passwords attempts are exceeded.

I'm closing this feature request as completed. Feel free to reopen it or create another issue if something's not working as expected.
Thanks!
